apis/kubedb/v1alpha1/perconaxtradb_helpers.go

```diff
@@ -202,6 +203,65 @@ func (p *PerconaXtraDBSpec) SetDefaults() {
 	if p.TerminationPolicy == "" {
 		p.TerminationPolicy = TerminationPolicyDelete
 	}
+	p.setDefaultProbes()
+}
+
+// setDefaultProbes sets defaults only when probe fields are nil.
+// In operator, check if the value of probe fields is "{}".
+// For "{}", ignore readinessprobe or livenessprobe in statefulset.
+// Ref: https://github.com/mattlord/Docker-InnoDB-Cluster/blob/master/healthcheck.sh#L10
+func (p *PerconaXtraDBSpec) setDefaultProbes() {
+	if p == nil {
+		return
+	}
+
+	pingCmd := []string{
+		"bash",
+		"-c",
+		`export MYSQL_PWD="${MYSQL_ROOT_PASSWORD}"
+ping_resp=$(mysqladmin -uroot ping)
+if [[ "$ping_resp" != "mysqld is alive" ]]; then
+    echo "[ERROR] server is not ready. PING_RESPONSE: $ping_resp"
+    exit 1
+fi
+`,
+	}
+	clusterCheckCmd := []string{
+		"/cluster-check.sh",
+	}
+
+	var readynessProbeCmd []string
+	if types.Int32(p.Replicas) > 1 {
+		readynessProbeCmd = clusterCheckCmd
+	} else {
+		readynessProbeCmd = pingCmd
+	}
+
+	readinessProbe := &core.Probe{
+		Handler: core.Handler{
+			Exec: &core.ExecAction{
+				Command: readynessProbeCmd,
+			},
+		},
+		InitialDelaySeconds: 30,
+		PeriodSeconds:       10,
+	}
+	if p.PodTemplate.Spec.ReadinessProbe == nil {
+		p.PodTemplate.Spec.ReadinessProbe = readinessProbe
+	}
+
+	livenessnessProbe := &core.Probe{
```

Comment:
Do we need liveness probes? We should not kill database when something goes wrong.

Reply:
Yes, there may have different reasons for happening something wrongs in database container. So, it is not wise to restart the pod without knowing the reason.

Removing ...